python-sync-actions/src/component.py

```diff
@@ -288,6 +288,35 @@ def find_array_property_path(response_data: dict, result_arrays: list = None) ->
 
         return result
 
+    def _add_page_params(self, job: Configuration, query_parameters: dict) -> dict:
+        """
+        Add page parameters to the query parameters
+        Args:
+            job: job configuration
+            query_parameters: query parameters
+
+        Returns:
+            query_parameters: updated query parameters
+        """
+
+        if job.api.pagination:
+            paginator_params = job.api.pagination.get(job.request_parameters.scroller)
+            if not paginator_params:
+                raise UserException(f"Paginator '{job.request_parameters.scroller}' not found in the configuration.")
+
+            if paginator_params.get("method") in ["offset", "pagenum"]:
+                pagination_method = get_pagination_method(paginator_params.get("method"))
+                paginator = pagination_method.add_page_params(paginator_params, query_parameters)
```

Comment:
Tady jsme si vubec neporozumeli bohuzel. Slo mi o "separation of concerns"

**config.py -> tridy a metody ktery drzi struktury konfigurace,.** zadna logika, ktera modifikuje query parametry tam nepatri
e.g. 
```
@dataclass
class PageNumPagination():
  limit:str
  limit_parameter:str = 'limit'
  page_parameter:str = 'page'
....
```
Pripadne to muze byt vic genericky jako u auth:

```
@dataclass
class Pagination(SubscriptableDataclass):
    type: str
    parameters: dict = field(default_factory=dict)
```

Tahle struktura pak bude zit uvnitr mapy paginations {key:Pagination}. Vsechny metody ktery se staraji o prevod z generic Ex struktury do tohodle jsou uvnitr configuration.py (jako treba u Auth metod).


**Pagination**

v idelanim svete bude paginator separatni objekt (viz jak to dela dlt). Mel by souviset s http klientem (jako authentication). Takze by mel vziknout novy modul uvnitr http_client `pagination.py` Uvnitr tohodle modulu bude builder podobne jako u auth method, ktera vyrobi paginator podle typu a parametru.

Kdyz pak zavolas client.make_call vrati to prvni stranku. Teda melo by to vracet iterator, protoze strankovani.



 Uvnitr component.py se chceme vyhnout vecem [typu](https://github.com/keboola/generic-extractor/pull/186/files#diff-7fff4f60bccf22855c889f4a657abdcc0ede1e2a8b08965eb9f62fdfb346cb60R311) ` if paginator_params.get("offsetFromJob"):` kde musis vedet jak se jmenuje ten parametr  `offsetFromJob`. co kdyz se ten nazev zmeni, co kdyz bude tech paginations vic a nektery nebudou mit offsetFromJob. Tohle by vubec nemela resit metodat uvnitr component.py, ale mel by si to resit ten paginator pro ktery je to specificky.



Takhle to sice funguje fajn, ale strasne blbe se to rozsiruje a kdybychom jednou chteli udelat svuj Generic v Pythonu (tj pouzit toho clienta) tak to bude dost narocny. Kdyz si nechame od zacatku ty veci modularni a oddeleny jak maji byt config=jenom user konfigurace, pagination=pagination, component=klient ktery vola http.client a orchestruje extrakci podle konfigu; tak to bude snadny zmenit. Nechci byt zbytecne hnidopisskej, ale kdyz budeme delat kompromisy, protoze je to ted nejrychlejsi workaround vsude, tak se nam to x krat vrati v budoucnu pri maintenance. Mozna na to muzeme sednout spolecne jeste